src/basic/terminal-util.h

```diff
@@ -64,6 +65,12 @@ typedef enum AcquireTerminalFlags {
 int acquire_terminal(const char *name, AcquireTerminalFlags flags, usec_t timeout);
 int release_terminal(void);
 
+int terminal_new_session(void);
```

Comment:
this helper is not used at all for now. which i guess ok if this is preparatory work for something else, but still i would feel more comfortable if we had a test for this.

maybe a simple test that allocates a pty, forks off a child, makes the child's ctty the peer of the pty, then checks that this works via get_ctty_devnr(), then detaches, verifies again, then maybe reattaches and checks again or so.